libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt

```diff
@@ -401,37 +399,27 @@ class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
+        unconditionalStatusPollingJob?.cancel()
         locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
         ioJobController.scope.launch {
             val currentDate = Date()
-            lastStatusUpdateTimeMillis = SystemClock.elapsedRealtime()
             navigator.updateLocation(rawLocation, currentDate)
             updateDataFromNavigatorStatus(currentDate)
         }
 
-        if (this.rawLocation == null) {
-            ioJobController.scope.launch {
-                while (isActive) {
-                    delay(STATUS_POLLING_INTERVAL)
-                    launch {
-                        emitUnconditionallyGetStatus()
-                    }
+        unconditionalStatusPollingJob = ioJobController.scope.launch {
+            delay(UNCONDITIONAL_STATUS_POLLING_PATIENCE)
+            while (isActive) {
+                launch {
+                    updateDataFromNavigatorStatus(Date())
                 }
+                delay(UNCONDITIONAL_STATUS_POLLING_INTERVAL)
             }
         }
 
         this.rawLocation = rawLocation
```

Comment:
Should we move this closer to where it's pushed L#403 `locationObservers.forEach { it.onRawLocationChanged(rawLocation) }`